tests/unittests/config/test_cc_ca_certs.py

```diff
@@ -62,30 +62,35 @@ def _fetch_distro(self, kind):
         paths = helpers.Paths({})
         return cls(kind, {}, paths)
 
-    def _mock_init(self):
+    def _mock_init(self, distro_name):
         self.mocks = ExitStack()
         self.addCleanup(self.mocks.close)
 
         # Mock out the functions that actually modify the system
         self.mock_add = self.mocks.enter_context(
             mock.patch.object(cc_ca_certs, "add_ca_certs")
         )
+        self.mock_remove = self.mocks.enter_context(
+            mock.patch.object(cc_ca_certs, "disable_default_ca_certs")
+        )
         self.mock_update = self.mocks.enter_context(
             mock.patch.object(cc_ca_certs, "update_ca_certs")
         )
-        self.mock_remove = self.mocks.enter_context(
-            mock.patch.object(cc_ca_certs, "remove_default_ca_certs")
-        )
+        if distro_name == "freebsd":
+            mock.patch(
```

Comment:
it doesn't make sense to add this mock here

Reply:
This function is common to all the testcases that are giving the subp error and so it made sense to put the mock in a single (shared) place.

Reply:
it doesn't make sense, because it's not consumed here.
the patch goes out of scope by the time and is destroyed